src/orders/components/OrderCustomerAddressesEditDialog/OrderCustomerAddressesEditDialog.tsx

```diff
@@ -127,132 +135,180 @@ const OrderCustomerAddressesEditDialog: React.FC<OrderCustomerAddressesEditDialo
 
   const countryChoices = mapCountriesToChoices(countries);
 
+  const [addressSearchState, setAddressSearchState] = React.useState<
+    OrderCustomerSearchAddressState
+  >({ open: false });
+
+  const parsedDefaultShippingAddress = validateDefaultAddress(
```

Comment:
I initially named the function `parseDefaultAddress` before deciding that validate is a better name here. I will change it to `validatedDefaultShippingAddress`

Reply:
I think a function that is called `validate` should `validate` - meaning it returns some form of an answer to a question if what we're passing is valid - whether it be a simple boolean or some more complicated structure. If this function actually parses the data, that means that the data we're providing is not valid or invalid - just different structure, types, etc. that we'd like to get. Which case is it then?

Reply:
I don't like using `parse` because of the fact that it can mean various things. When I see `parse` in code I only know that this function takes some data and changes the shape of it. Many times it can be named better.

In this particular situation function checks if the data is valid, because there might be no default address on the backend or it might be wrong - then we return a fallback address. Thus the default address is `validated`, but its shape is not changed as `parsed` would suggest.